.github/workflows/pythonapp.yml

```diff
@@ -1,19 +1,52 @@
-name: Python application
+name: "python tests+artifacts+release"
 
 on:
   pull_request:
   push:
-    branches: [main]
+    branches:
+    - main
+    tags:
+    - "v*"
+  release:
+    types: [published]
+
 
 jobs:
+  dist:
+    runs-on: ubuntu-latest
+    name: Python sdist/wheel
+    steps:
+    - uses: actions/checkout@v1
+    - uses: actions/setup-python@v2
+      with:
+        python-version: "3.9"
+    - name: Install dependencies
+      run: |
+        python -m pip install --upgrade pip
+        pip install --upgrade wheel setuptools build
+
+    - name: Build package
+      run: python -m build -o dist/
+    - uses: actions/upload-artifact@v2
+      with:
+        name: dist
+        path: dist
+
   test:
     runs-on: ubuntu-latest
+    needs: [dist]
     strategy:
       matrix:
         # todo: extract from source
-        python-version: [2.7, 3.4, 3.5, 3.6, 3.7, 3.8, 3.9]
```

Comment:
@RonnyPfannschmidt was the removal of 3.4 accidental? This caused https://github.com/pytest-dev/apipkg/issues/29 bug. It is still possible to test it if there's need. But only under ubuntu-18.04: https://github.com/actions/python-versions/blob/c483657/versions-manifest.json#L1228.
Here's an example of such an inclusion: https://github.com/cherrypy/cheroot/commit/b332301ef0670445e67e9167697d437fdd385fc9#diff-207f09dfa0bc604a684631a1c78be7d9aa00f9a6c778eec30afa287990be3da1R32-R36.